repl/src/main/scala/com/cloudera/livy/repl/SparkInterpreter.scala

```diff
@@ -84,11 +87,33 @@ class SparkInterpreter(conf: SparkConf) extends Interpreter {
     setContextClassLoaderMethod.invoke(sparkIMain)
 
     sparkContext = SparkContext.getOrCreate(conf)
+    var sqlContextClassName = "org.apache.spark.sql.SQLContext"
+    try {
+      val loader = Option(Thread.currentThread().getContextClassLoader)
+        .getOrElse(getClass.getClassLoader)
+      val sqlContextClassName = "org.apache.spark.sql.hive.HiveContext"
+      sqlContext = loader.loadClass(sqlContextClassName).getConstructor(classOf[SparkContext])
+        .newInstance(sparkContext).asInstanceOf[SQLContext]
+      info("Created sql context (with Hive support)..")
+    } catch {
+      case _: java.lang.ClassNotFoundException | _: java.lang.NoClassDefFoundError =>
+        sqlContext = new SQLContext(sparkContext)
+        info("Created sql context..")
+      case _ if sys.env.getOrElse("LIVY_TEST", "false") == "true" =>
```

Comment:
Also why would you get an exception here during tests? The hive code is there, it should work.